src/controllers/payment.controller.ts

```diff
@@ -0,0 +1,144 @@
+import {
+  createStripeSession,
+  checkPaymentStatus,
+  deleteSession,
+} from '../utils/stripe';
+import dotenv from 'dotenv';
+import { Request, Response } from 'express';
+import PaymentService from '../service/payment.service';
+import { PaymentStatus } from '../models/payment.model';
+import Stripe from 'stripe';
+
+dotenv.config();
+
+async function makepaymentsession(req: CustomRequest, res: Response) {
+  const user = req.user;
+  const { orderId } = req.body;
+  if (!user) {
+    return res.status(403).json({ message: 'Unauthorized' });
+  }
+
+  const baseUrl = process.env.BASE_URL || '';
+  const success_url = `${baseUrl}/api/v1/payment/success?userId=${user.user_id}&orderId=${orderId}`;
+  const cancel_url = `${baseUrl}/api/v1/payment/cancel?userId=${user.user_id}&orderId=${orderId}`;
+
+  try {
+    const lineItems = [
+      {
+        price_data: {
+          currency: 'usd',
+          product_data: {
+            name: 'T-shirt',
+          },
+
+          unit_amount: 90,
+        },
+
+        quantity: 1,
```

Comment:
If these are the items we're paying for, will we have to hardcode them here? Or they should come from the order? Also what if the order has many items? Brainstorm on that.

Reply:
for this products which are hardcoded is a demonstration of how the person who will implementing checkout will be using payment and it flexible you can add any number of products.